airflow/models/dag.py

```diff
@@ -734,15 +759,22 @@ def task_group(self) -> "TaskGroup":
 
     @property
     def filepath(self) -> str:
-        """File location of where the dag object is instantiated"""
-        fn = self.full_filepath.replace(settings.DAGS_FOLDER + '/', '')
-        fn = fn.replace(os.path.dirname(__file__) + '/', '')
+        """:meta private:"""
+        warnings.warn(
+            "filepath is deprecated, use relative_fileloc instead", DeprecationWarning, stacklevel=2
+        )
+        return self.relative_fileloc
+
+    @property
+    def relative_fileloc(self) -> str:
+        """File location of the importable dag 'file' relative to the configured DAGs folder."""
```

Comment:
How will this field behave for example DAGs?

Reply:
In this case it would be `/usr/lib/python3.7/.../airflow/example_dags/example_bash_operator.py` -- so not relative at all.

I'll update/expand the docstring to say how it deals with DAGs outside of the dags folder.

Reply:
Maybe we are also able to handle examples DAGs by adding a suffix? This will allow us to install workers on shared machines without any problems
For samples:
`[exaample_dags_folder]/example_bash_operator.py` => `/usr/lib/python3.7/.../airflow/example_dags/example_bash_operator.py`
`[dags_folder]/example_bash_operator.py` => `~/home/airflow/dags`

Reply:
I think this change would be better done along with something like https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-20+DAG+manifest, so I've not done this for now.

Reply:
Maybe we shouldn’t call this *relative* fileloc, but just fileloc…? Not sure about this.